### ISSUE TITLE
Use of unmaintained crates

### ISSUE THREAD

**Issue by myOmikron** [tectonic-typesetting/tectonic#1034]
Hello there,

first of all, thanks for this amazing project.

`cargo-deny` reports unmaintained + unsound crates:

```
error[unmaintained]: `net2` crate has been deprecated; use `socket2` instead
    ┌─ /home/omikron/git/reporter/Cargo.lock:230:1
    │
230 │ net2 0.2.38 registry+https://github.com/rust-lang/crates.io-index
    │ ----------------------------------------------------------------- unmaintained advisory detected
    │
    = ID: RUSTSEC-2020-0016
    = Advisory: https://rustsec.org/advisories/RUSTSEC-2020-0016
    = The [`net2`](https://crates.io/crates/net2) crate has been deprecated
      and users are encouraged to considered [`socket2`](https://crates.io/crates/socket2) instead.
    = Announcement: https://github.com/deprecrated/net2-rs/commit/3350e3819adf151709047e93f25583a5df681091
    = Solution: No safe upgrade is available!
    = net2 v0.2.38
      ├── mio v0.6.23
      │   ├── mio-extras v2.0.6
      │   │   └── notify v4.0.17
      │   │       └── watchexec v1.17.2
      │   │           └── tectonic v0.12.0
      │   │               └── reporter v0.1.0
      │   └── notify v4.0.17 (*)
      └── miow v0.2.2
          └── mio v0.6.23 (*)

error[unmaintained]: ansi_term is Unmaintained
   ┌─ /home/omikron/git/reporter/Cargo.lock:30:1
   │
30 │ ansi_term 0.12.1 registry+https://github.com/rust-lang/crates.io-index
   │ ---------------------------------------------------------------------- unmaintained advisory detected
   │
   = ID: RUSTSEC-2021-0139
   = Advisory: https://rustsec.org/advisories/RUSTSEC-2021-0139
   = The maintainer has advised that this crate is deprecated and will not receive any maintenance.
     
     The crate does not seem to have much dependencies and may or may not be ok to use as-is.
     
     Last release seems to have been three years ago.
     
     ## Possible Alternative(s)
     
      The below list has not been vetted in any way and may or may not contain alternatives;
     
      - [anstyle](https://github.com/epage/anstyle)
      - [console](https://crates.io/crates/console)
      - [nu-ansi-term](https://crates.io/crates/nu-ansi-term)
      - [owo-colors](https://crates.io/crates/owo-colors)
      - [stylish](https://crates.io/crates/stylish)
      - [yansi](https://crates.io/crates/yansi)
     
     ## Dependency Specific Migration(s)
     
      - [structopt, clap2](https://github.com/clap-rs/clap/discussions/4172)
   = Announcement: https://github.com/ogham/rust-ansi-term/issues/72
   = Solution: No safe upgrade is available!
   = ansi_term v0.12.1
     └── clap v2.34.0
         └── structopt v0.3.26
             └── tectonic v0.12.0
                 └── reporter v0.1.0

warning[unsound]: Potential unaligned read
   ┌─ /home/omikron/git/reporter/Cargo.lock:42:1
   │
42 │ atty 0.2.14 registry+https://github.com/rust-lang/crates.io-index
   │ ----------------------------------------------------------------- unsound advisory detected
   │
   = ID: RUSTSEC-2021-0145
   = Advisory: https://rustsec.org/advisories/RUSTSEC-2021-0145
   = On windows, `atty` dereferences a potentially unaligned pointer.
     
     In practice however, the pointer won't be unaligned unless a custom global allocator is used.
     
     In particular, the `System` allocator on windows uses `HeapAlloc`, which guarantees a large enough alignment.
     
     # atty is Unmaintained
     
     A Pull Request with a fix has been provided over a year ago but the maintainer seems to be unreachable.
     
     Last release of `atty` was almost 3 years ago.
     
     ## Possible Alternative(s)
     
     The below list has not been vetted in any way and may or may not contain alternatives;
     
      - [is-terminal](https://crates.io/crates/is-terminal)
      - std::io::IsTerminal *nightly-only experimental*
   = Announcement: https://github.com/softprops/atty/issues/50
   = Solution: No safe upgrade is available!
   = atty v0.2.14
     ├── clap v2.34.0
     │   └── structopt v0.3.26
     │       └── tectonic v0.12.0
     │           └── reporter v0.1.0
     └── tectonic v0.12.0 (*)
```
     
As of `cargo 1.69.0 (6e9a83356 2023-04-12)`, the compiler also reports this warning:

```
warning: the following packages contain code that will be rejected by a future version of Rust: quick-xml v0.22.0
```

For those, which use the project as library, being able to use `cargo-deny` for checking their dependencies may be quite important. 

Thanks!

**Comment by pkgw** [comment]
As it happens, #1033 was just filed and should address many if not all of these issues!

**Comment by myOmikron** [comment]
Sounds great, I'll check if the PR is merged, if any of the states problems exist anymore, and close this issue accordingly

**Comment by CraftSpider** [comment]
The clap dependency wasn't changed, but it's test only. I can make a new PR for it if desired after the current one is merged.

**Comment by myOmikron** [comment]
The `net2` and `quick-xml` ones are resolved (checked v0.13.0)
